src/pages/productListing/ProductList.js

```diff
@@ -5,39 +5,37 @@ import Card from "../../components/Card/Card";
 import React from "react";
 import Navbar from "../../components/Navbar/Navbar";
 import FilterSideBar from "../../components/FilterSideBar/FilterSideBar";
+import useAxios from "../../CustomHooks/useAxios";
+import { useProducts } from "../../Context/product-context";
 
 const ProductList = () => {
-  const [products, setProducts] = useState([]);
+  useAxios("/api/products", "get");
 
+  const [products, setProducts] = useState("");
```

Comment:
Have your useState at the top of your function because in React all of your hooks should be at the top of your function, otherwise it is an antipattern.